File: SortingAlgorithmsCpp/SortingAlgorithms/QuickSort.cpp
#include "QuickSort.hpp"
#include <iostream>

int partition(int arr[], int f, int l) {
	int pivot = arr[l];

	int i = f - 1;

	for (int j = f; j < l; j++) {
		if (arr[j] <= pivot) {
			i++;
			std::swap(arr[j], arr[i]);
		}
	}
	std::swap(arr[i + 1], arr[l]);

	return i + 1;
}

void quickSort(int arr[], int f, int l) {
	if (f < l) {
		int pivot = partition(arr, f, l);
		quickSort(arr, f, pivot - 1);
		quickSort(arr, pivot, l);
	}
}

File: SortingAlgorithmsPython/quickSort.py
def partition(arr, f, l):
    pivot = arr[l]

    i = f - 1

    for j in range(f, l):
        if arr[j] <= pivot:
            i += 1
            arr[j], arr[i] = arr[i], arr[j]

    arr[i + 1], arr[l] = arr[l], arr[i + 1]

    return i + 1


def quick_sort(arr, f, l):
    if f < l:
        pivot = partition(arr, f, l)
        quick_sort(arr, f, pivot - 1)
        quick_sort(arr, pivot, l)